### PR TITLE
feat(term): termios package

### PR DESCRIPTION
**Pull request by caarlos0**
This will help us remove the dep on u-root/u-root from /ssh, and also adds support to netbsd.

The api is different, and it does not use reflection.

### CODE REVIEW

**Review by aymanbagabas — COMMENTED** [review]
Noice! Does this build on all BSDs? Should we support `dragonfly`? Would be nice to have a workflow to build for all these OSes

**Comment by caarlos0** [comment]
Assuming dragonfly uses the same syscalls, its possible yeah.

agreed, wouldn't hurt to `go build` for all `GOOS/GOARCH` to see if it at least compiles.

File: exp/term/termios/termios.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	"ocrnl":  {O, syscall.OCRNL},
+	"onocr":  {O, syscall.ONOCR},
+	"onlret": {O, syscall.ONLRET},
+
+	"cs7":    {C, syscall.CS7},
+	"cs8":    {C, syscall.CS8},
+	"parenb": {C, syscall.PARENB},
+	"parodd": {C, syscall.PARODD},
+
+	// XXX: not available on some OSs
```
Comment (aymanbagabas) [inline exp/term/termios/termios.go:156]: We can define these for the OSs that support them in a `init`

File: exp/term/termios/termios.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+const (
+	I = iota // Input control
+	O        // Output control
+	C        // Control
+	L        // Line control
+)
+
+// https://www.man7.org/linux/man-pages/man3/termios.3.html
+var allCcOpts = map[string]int{
+	"intr":    syscall.VINTR,
```
Comment (aymanbagabas) [inline exp/term/termios/termios.go:96]: Should we define the strings as consts?

Comment (caarlos0) [reply exp/term/termios/termios.go:96]: was thinking about it too...

**Pull request merged**